File: apps/api/app/archive_knowledge/contracts/impact.py
from __future__ import annotations

from pydantic import BaseModel, Field


class ImpactSet(BaseModel):
    impact_set_id: str
    archive_id: str
    policy_package_version_id: str
    previous_policy_package_version_id: str
    changed_rule_ids: list[str] = Field(default_factory=list)
    affected_stage_ids: list[str] = Field(default_factory=list)
    affected_chunk_ids: list[str] = Field(default_factory=list)
    affected_candidate_ids: list[str] = Field(default_factory=list)
    minimum_rebuild_stage_id: str
    affected_document_ids: list[str] = Field(default_factory=list)
    affected_object_ids: list[str] = Field(default_factory=list)
    affected_relation_ids: list[str] = Field(default_factory=list)
    affected_publication_snapshot_ids: list[str] = Field(default_factory=list)
    requires_governance_reconfirmation: bool = False
    generated_at: str | None = None
    writes_official_knowledge: bool = False

File: apps/web/src/features/p1/contracts/impact.ts
export interface ImpactSet {
  impact_set_id: string;
  archive_id: string;
  policy_package_version_id: string;
  previous_policy_package_version_id: string;
  changed_rule_ids: string[];
  affected_stage_ids: string[];
  affected_chunk_ids?: string[];
  affected_candidate_ids?: string[];
  minimum_rebuild_stage_id: string;
  affected_document_ids: string[];
  affected_object_ids: string[];
  affected_relation_ids: string[];
  affected_publication_snapshot_ids: string[];
  requires_governance_reconfirmation?: boolean;
  generated_at?: string;
  writes_official_knowledge: boolean;
}
